pennylane/tape/qnode.py

```diff
@@ -410,17 +420,6 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
-
-        if self.interface == "autograd":
-            # HOTFIX: to maintain compatibility with core, here we treat
-            # all inputs that do not explicitly specify `requires_grad=False`
-            # as trainable. This should be removed at some point, forcing users
-            # to specify `requires_grad=True` for trainable parameters.
-            args = [
-                anp.array(a, requires_grad=True) if not hasattr(a, "requires_grad") else a
-                for a in args
-            ]
-

```

Comment:
I am wondering about this change? Isn't the tape constructed each time the qnode is called?